README.md

```diff
@@ -1,31 +1,348 @@
 # Hyperion
 
-TODO: Write a gem description
+Hyperion is a Ruby REST client that follows certain conventions
+layered on top of HTTP. The conventions implement best practices
+surrounding versioning, error handling, and crafting an API for
+consumption by third parties. Hyperion provides abstractions that
+make it easy to follow these conventions consistently across
+projects.
 
-## Installation
+This document describes the conventions, then demonstrates how to
+use the API, and finally shows how to test your client-side code.
 
-Add this line to your application's Gemfile:
+## Conventions
+
+### Versioning
+
+_TODO_
+
+### Errors
+
+400-level errors are always returned as exactly 400; no distinction is
+made in the error code. Instead, a well-defined structure is returned
+that contains
+
+- a human-oriented error message, and
+- a machine-oriented list of `ErrorInfo`s.
+
+The point of the message is to describe the problem. The point of the
+error infos is to provide enough information to begin resolving the
+problem.
+
+An `ErrorInfo` consists of:
+
+- code
+- resource
+- field
+- value
+- reason
+
+As of this writing, allowable codes are
+
+- 'missing'
+- 'missing_field'
+- 'invalid'
+- 'unsupported'
+- 'already_exists'
+
```

Comment:
Yeah, I debated duplicating these here because of the sync problem. I concluded it was worth it. I really don't want people to have to jump into code after less than a page of the readme. Plus the link would be just as brittle as duplicating them here. (I tend to reorganize source files.) Hyperion is expected to be a stable, depended-upon component, so sync problems will dwindle as hyperion becomes more stable.